src/intervals/rounding.jl

```diff
@@ -66,75 +103,99 @@ function split_interval_string(T, x::AbstractString)
-
-
+    @controlled_round(T, parse(T, m.captures[1]), parse(T, m.captures[2]))
 end
 
 
 @doc doc"""`make_interval` is used by `@interval` to create intervals from
 individual elements of different types""" ->
-make_interval(::Type{BigFloat}, x::AbstractString)    =  split_interval_string(BigFloat, x) #@thin_round(BigFloat, parse(BigFloat,x))
-make_interval(::Type{BigFloat}, x::Irrational) =  @thin_round(BigFloat, big(x))
-make_interval(::Type{BigFloat}, x::Integer)   =  Interval(BigFloat(x))  # no rounding -- dangerous if very big integer
-# but conversion from BigInt to BigFloat with correct rounding seems to be broken anyway # @thin_interval(BigFloat("$x"))
 
-make_interval(::Type{BigFloat}, x::Rational)  =  make_interval(BigFloat, x.num) / make_interval(BigFloat, x.den)
+# make_interval for BigFloat intervals
+make_interval(::Type{BigFloat}, x::AbstractString) =
+    split_interval_string(BigFloat, x)
+make_interval(::Type{BigFloat}, x::Irrational) = @thin_round(BigFloat, big(x))
+function make_interval(::Type{BigFloat}, x::Rational)
+    isinf(x) && return Interval(convert(BigFloat,x))
+    make_interval(BigFloat, x.num) / make_interval(BigFloat, x.den)
+end
 function make_interval(::Type{BigFloat}, x::Float64)
     isinf(x) && return Interval(convert(BigFloat,x))
-    make_interval(BigFloat, rationalize(x))  # NB: converts a float to a rational -- this could be slow
+    split_interval_string(BigFloat, string(x))
+end
+make_interval(::Type{BigFloat}, x::Integer) =
+    @thin_round(BigFloat, convert(BigFloat, x))
+make_interval(::Type{BigFloat}, x::BigFloat) = @thin_round(BigFloat, x)  # convert to possibly different BigFloat precision
+function make_interval(::Type{BigFloat}, x::Interval)
+    a = make_interval(BigFloat, x.lo)
+    b = make_interval(BigFloat, x.hi)
+    Interval(a.lo, b.hi)
 end
 
-make_interval(::Type{BigFloat}, x::BigInt)    =  @thin_round(BigFloat, convert(BigFloat, x))
-
-make_interval(::Type{BigFloat}, x::BigFloat)  =  @thin_round(BigFloat, 1.*x)  # convert to possibly different BigFloat precision
-make_interval(::Type{BigFloat}, x::Interval)  =  @round(BigFloat, big(1.*x.lo), big(1.*x.hi))
-
-
-make_interval(::Type{Float64}, x::AbstractString)    =  split_interval_string(Float64, x) #@thin_round(Float64, parse(Float64, x))
-make_interval(::Type{Float64}, x::Irrational) =  make_interval(Float64, make_interval(BigFloat, x))
 
-make_interval(::Type{Float64}, x::Integer)   =  Interval(float(x))    # assumes the int is representable
-make_interval(::Type{Float64}, x::Rational)  =  make_interval(Float64, x.num) / make_interval(Float64, x.den)
+# make_interval for Float64 intervals
+make_interval(::Type{Float64}, x::AbstractString) = split_interval_string(Float64, x)
+function make_interval(::Type{Float64}, x::Irrational)
+    a = with_bigfloat_precision(53) do
+        make_interval(BigFloat,x)
+    end
+    float(a)
+end
+function make_interval(::Type{Float64}, x::Rational)
```

Comment:
Yet, I just noticed that `@interval(1//0)` and `make_interval(Float64,1//0)` yield different numerical results, namely, `∅` and `[∞, ∞]`, respectively. The former is not correct (1//0 corresponds to `Inf` for `Rational`). Similar things occur with `@biginterval` and `make_interval(BigFloat, 1//0)`.

This is related to `transform`, which calls the actual operations among intervals, which is yields the `∅` if the interval in the denominator is the zero interval. I don't think this is too tragic...